lib/jefferies_tube/capistrano/rails.rb

```diff
@@ -3,24 +3,32 @@
   task :console do
     console_roles = [roles(:worker), roles(:app)].find(&:any?)
     on console_roles, primary: true do |host, user|
-      rails_env = fetch(:rails_env)
       run_interactively "RAILS_ENV=#{rails_env} bundle exec rails console"
     end
   end
 
   desc "Open the rails dbconsole on each of the remote servers"
   task :dbconsole do
     on roles(:db), primary: true do |host|
-      rails_env = fetch(:rails_env)
       run_interactively "RAILS_ENV=#{rails_env} bundle exec rails dbconsole"
     end
   end
 
   desc "Open the rails log"
   task :log do
     on roles(:app), primary: true do |host, user|
-      rails_env = ENV['LOG'] || fetch(:rails_env)
-      run_interactively %Q{tail -f log/#{rails_env}.log | grep --line-buffered --invert "Delayed::Backend::ActiveRecord::Job Load"}
+      filename = ENV['LOG'] || rails_env
+      run_interactively %Q{tail -f log/#{filename}.log | grep --line-buffered --invert "Delayed::Backend::ActiveRecord::Job Load"}
+    end
+  end
+
+  desc "Run a rake task"
+  task :rake, :task_to_run do |_, parameters|
+    task_to_run = parameters[:task_to_run]
+    abort "Must supply task to run on remote server.\nUsage example: cap dev rails:rake[flipper:synchronize_features]" if task_to_run.nil?
+    roles = [roles(:worker), roles(:app)].find(&:any?)
```

Comment:
what's this line of code do?

Reply:
this is some pretty dense ruby, but it looks like it's making an array of those 2 things then `find` is finding the first thing that meets the predicate `&:any?`. so the first thing where `any?` returns true

Reply:
Yep, it first checks to see if there are any workers and if not (any? returns false), then (there should be as is our normal convention) we use the app role.